web/src/components/PreflightResultErrors.tsx

```diff
@@ -0,0 +1,239 @@
+import React, { useEffect, useReducer } from "react";
+import { useRouteMatch } from "react-router";
+import MonacoEditor from "@monaco-editor/react";
+import CodeSnippet from "./shared/CodeSnippet";
+import ErrorModal from "./modals/ErrorModal";
+import { Utilities } from "../utilities/utilities";
+import "../scss/components/PreflightCheckPage.scss";
+
+import { KotsParams } from "@types";
+
+type Props = {
+  ignorePermissionErrors: () => void;
+  logo: string;
+  preflightResultData: PreflightResultData | null;
+  valueFromAPI: string;
+};
+
+type State = {
+  command: string | null;
+  displayErrorModal: boolean;
+  errorTitle: string;
+  errorMsg: string;
+  showErrorDetails: boolean;
+};
+
+type PreflightResultData = {
+  appSlug: string;
+  sequence: number;
+};
+// class PreflightResultErrors extends Component {
+const PreflightResultErrors = (props: Props) => {
+  const [state, setState] = useReducer(
+    (currentState: State, newState: Partial<State>) => ({
+      ...currentState,
+      ...newState,
+    }),
+    {
+      command: null,
+      showErrorDetails: false,
+      errorTitle: "",
+      errorMsg: "",
+      displayErrorModal: false,
+    }
+  );
+  const match = useRouteMatch<KotsParams>();
```

Comment:
added `useRouteMatch` hook to replace `withRouter` props

Reply:
cool. never knew about this hook